src/aolayer.cpp

```diff
@@ -16,7 +17,11 @@ AOLayer::AOLayer(QWidget *p_parent, AOApplication *p_ao_app) : QLabel(p_parent)
   ticker = new QTimer(this);
   ticker->setTimerType(Qt::PreciseTimer);
   ticker->setSingleShot(false);
-  connect(ticker, SIGNAL(timeout()), this, SLOT(movie_ticker()));
+  // we can't connect the timer to the ticker function yet because we have to
+  // let CharLayer connect to its own ticker function and if we do both the
+  // ticker will fire twice which breaks things. instead we have to let each
+  // subclass connect to its own ticker function, and since none of the rest of
+  // them have one, they all connect to the superclass ticker function instead
```

Comment:
You need to make `movie_ticker` virtual. That is where your anguish is coming from.

Reply:
thank you